app/backend/src/app/auth/auth.service.ts

```diff
@@ -81,5 +95,63 @@ export class AuthService {
     };
   }
 
-  
+  async signInWithEmail(signInDto: SignInDto): Promise<{ student: Partial<Student>; token: string }> {
+    const { email, password } = signInDto;
+    const student = await this.studentRepository.findOne({ where: { email } });
+
+    if (!student) {
+      throw new UnauthorizedException('Invalid email or password');
+    }
+
+    const isPasswordValid = await bcrypt.compare(password, student.password);
+    if (!isPasswordValid) {
+      throw new UnauthorizedException('Invalid email or password');
+    }
+
+    const token = jwt.sign({ id: student.id, email: student.email }, this.jwtSecret, { expiresIn: '1h' });
+
+    return {
+      student: {
+        id: student.id,
+        first_name: student.first_name,
+        last_name: student.last_name,
+        email: student.email,
+        image_url: student.image_url,
+        isActive: student.isActive,
+      },
+      token
+    };
+  }
+
+  async signInWithGoogle(signInDto: SignInDto): Promise<{ student: Partial<Student>; token: string }> {
+    const { googleId } = signInDto;
+
+    // Find all students with external_auth true
+    const students = await this.studentRepository.find({ where: { external_auth: true } });
+
+    // Check if any student's auth_pass matches the hashed Google ID
+    for (const student of students) {
```

Comment:
we dont need to do this we just get one user with same uuid which user have provided us in the dto , then we have the user data in out variable and from that data we can get the authPass and compare it with the googleId